project/src/Lib.hs

```diff
@@ -1,4 +1,191 @@
 module Lib where
+import Data.Char
 
-main :: IO ()
-main = putStrLn "Hi!"
+type Dict a b = [(a, b)]
+
+data Tree a = Null | Node a (Tree a) (Tree a)
+    deriving (Eq, Show)
+
+makeTree :: String -> Tree Int
+makeTree "" = Null
+makeTree (c0:r) = 
+  if c0 == 'N' 
```

Comment:
You should be using an abstract data type for this. What does `'N'` mean, what does `'u'` mean?

Reply:
'N' means we either have "Node" or "Null". If we have Null, then we create an empty tree. If it is "Node", then it's followed by a numerical value, which is extrapolated by the `getValue` function. Forgot to mention, the `makeTree` function expects only legitimate input (the string representation of an actual tree).